akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorContextAdapter.scala

```diff
@@ -120,11 +119,8 @@ private[akka] object ActorContextAdapter {
     ActorRefAdapter[U](ref)
   }
 
-  private def initLoggerWithClass(logClass: Class[_]): LoggerAdapterImpl = {
-    val logSource = self.path.toString
-    val system = untypedContext.system.asInstanceOf[ExtendedActorSystem]
-    val logger =
-      new LoggerAdapterImpl(system.eventStream, logClass, logSource, LoggingFilterWithMarker.wrap(system.logFilter))
+  private def initLoggerWithClass(logClass: Class[_]): Logger = {
+    val logger = LoggerFactory.getLogger(logClass)
```

Comment:
We also want to associate the logger with the actor path some how, that's what the `logSource` previously did.

Reply:
How you find concatenation with '/' or some other separator?

Reply:
It would be great if it would be possible to have it as a separate field, so when published to log infrastructure that supports structured logs (like Kibana) it is easy to group/filter/report on it

Reply:
I wonder if MDC is the only option, that kind of sucks because it would mean having to always setmdc/touch thread local when actor starts running and clear after.

Reply:
I agree that it should be a separate field and not part of the Logger name. The Logger name should be plain class names, since that is the normal convention with slf4j.
Not sure what possibilities there are for adding "fields"? Is that with MDC, Markers or something else?
Maybe  @wsargent can give us a hint?

Markers: https://tersesystems.com/blog/2019/05/18/application-logging-in-java-part-4/

Reply:
@patriknw There are two ways you can do this using markers.  You'll always end up doing something like:

```scala
val customPathMarker: PathMarker = PathMarker(actor.path)
logger.info(customPathMarker, message)
```

but you can do that either directly, or by specifying a proxy logger that will always add the context information under the hood.

```scala
val proxyLogger: org.slf4j.Logger = PathLoggerFactory(loggerFactory, actor.path).getLogger(getClass())
proxyLogger.info(message) // will automatically add a path marker under the hood.
```

I'm still writing out the fine details, but check out 

https://github.com/tersesystems/terse-logging-apis/blob/master/logging-example-context/src/main/java/com/tersesystems/logging/example/context/AppLogger.java

and

https://github.com/tersesystems/terse-logging-apis/blob/master/logging-context-logstash/src/main/java/com/tersesystems/logging/context/logstash/AbstractLogstashLogger.java

the most general case is

https://github.com/tersesystems/terse-logging-apis/blob/master/logging-api-proxy/src/main/java/com/tersesystems/logging/api/proxy/ProxyLogger.java

I generally create these with Javapoet as it's a pretty large API:

https://github.com/tersesystems/terse-logging-apis/tree/master/logging-api-generator

Reply:
thanks wsargent, we will look into that. Very useful if we can include some fields in all logging (from Loggers created by Akka)

Reply:
If you're specifically interested in the actor path, you may want to create the logger based on the actorsystem + path:

```scala
val convertedPath = convertToDotFormat(actorSystem + actor.path)
val logger loggerFactory.getLogger(convertedPath)
```

You're not limited to using the class name as the logger name, it's just what most people default to.

Reply:
I think the problem is that we want both

Reply:
I think most users want the Logger name to correspond to package/class name (because that is what they are used to). Actor path is a good additional field but shouldn't be for the logger name.

Reply:
Outstanding info guys, thanks! 
From what I understand from above looks to me Marker is the cleaner solution? Shall I go for it?

I've been looking around regards when to use MDC vs Marker and I got from it that MDC is better suited to slice and dice or to aggregate info while Markers are more focused for filtering info, even before gets written into the log or to trigger events when some mark is found. 
Some of this info I got it from https://stackoverflow.com/questions/4165558/best-practices-for-using-markers-in-slf4j-logback which actually I found is also referred  in https://github.com/akka/akka/pull/27324#discussion_r302697422

My only concern regards using MDC, in this case, would be it looks a bit more cumbersome. But I'm far from an expert in logging. I will be fine going, either way by now, MDC or Marker, as I don't have a strong opinion about it. Please let me know what you find is best.

Reply:
From my limited knowledge MDC sounds more right as it is context rather than something special with an event (like how we use markers for "Security" previously for example).

Reply:
I'm inclined to MDC as well.

Reply:
@johanandren @patriknw 
I'm a bit confused about this part, to be honest.

Let's say I add the actorPath when init the logger in this very same class: ``` 
private def initLoggerWithClass(logClass: Class[_]): Logger = {
    val logger = LoggerFactory.getLogger(logClass)
    org.slf4j.MDC.put("actorPath",self.path.name)
    actorLogger = OptionVal.Some(logger)
    logger
  }
```

What I don't quite get is at what point will I clean it. 
If (big if) I do it on akka.actor.typed.internal.adapter.ActorAdapter in .... preStart? postStop?
looks I'm splitting functionality that should be in the same place, related in a way that is easy to follow. On the contrary, this will make the code difficult to understand and maintain. 
Not only that but I'm not even sure that would fulfill the requirements. I will have to write down a test for this as I understand there's nothing currently that is testing this. 

Any suggestions are very much welcome.